### ISSUE TITLE
Switch to docopt

### ISSUE THREAD

**Issue by adamralph** [scriptcs/scriptcs#948]
For consideration...

http://docopt.org/ is a cross platform initiative to unify command line syntax (thanks to @BrendanThompson for putting me onto this in https://github.com/scriptcs-contrib/svm/issues/7).

There is an official .NET port at https://github.com/docopt/docopt.net.

It's breaking because it would mean switching to double dashes for long form options.

**Comment by filipw** [comment]
what is the advantage?

**Comment by adamralph** [comment]
Uniformity with other apps across all platforms.

**Comment by adamralph** [comment]
BTW - in my refactoring PR (https://github.com/scriptcs/scriptcs/pull/944) command line args have become decoupled from the command factory so a switch would be easier.

I raised this just for consideration and marked it P3. Certainly not a priority but may be nice to align scriptcs with a common standard at some point.

**Comment by filipw** [comment]
But is this a standard? If one self-declares being a standard it doesn't mean anything yet ;)
A quick google shows it's not even a standard in Python world, not to mention other languages.

Don't get me wrong, I am not trolling, I am genuinely interested in what are we aligning with here

**Comment by adamralph** [comment]
Fair enough. The 'standard' itself should be ratified before going ahead. Marked as 'hold'.

**Comment by adamralph** [comment]
One thing that certainly is impressive is the amount of platforms it has been ported to. Current count is 20. It's the closest thing to a CLI standard I've seen so far.

https://github.com/docopt

**Comment by BrendanThompson** [comment]
Docopt is a pretty neat little library. I'm pretty sure I suggested moving to double dashes a long time ago :). Mostly to align with the Linux/Unix world. You can also retain the short form single dash.

**Comment by c0ntinuum** [comment]
+1 (e.g -v or --version)
